libraries/python/chat-driver/chat_driver/openai_chat_completion_driver.py

```diff
@@ -233,9 +272,9 @@ async def respond(
 
         completion_args = {
             "messages": (
-                self._format_instructions(self.messages, instruction_parameters)
+                self._format_instructions(self.messages_provider.get(), instruction_parameters)
                 if instruction_parameters
-                else self.messages
+                else self.messages_provider.get()
```

Comment:
here's some more smell that the instruction message should be kept separate

Reply:
Yeah. Makes sense. Let me know what you think of the new way.